azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -195,87 +213,94 @@ def _fetch_goal_state(wire_client):
             try:
-                vm_settings, vm_settings_updated = self._wire_client.get_host_plugin().fetch_vm_settings(force_update=force_update)
+                try:
+                    vm_settings, vm_settings_updated = wire_client.get_host_plugin().fetch_vm_settings()
+                except ResourceGoneError:
+                    # retry after refreshing the HostGAPlugin
+                    GoalState.update_host_plugin_headers(wire_client)
+                    vm_settings, vm_settings_updated = wire_client.get_host_plugin().fetch_vm_settings()
 
             except VmSettingsNotSupported:
                 pass
-            except VmSettingsError as exception:
-                # ensure we save the vmSettings if there were parsing errors
-                self._history.save_vm_settings(ExtensionsGoalStateFromVmSettings.redact(exception.vm_settings_text))
+            except VmSettingsParseError as exception:
+                # ensure we save the vmSettings if there were parsing errors, but save them only once per ETag
+                if not GoalStateHistory.tag_exists(exception.etag):
+                    GoalStateHistory(timeutil.create_timestamp(), exception.etag).save_vm_settings(exception.vm_settings_text)
                 raise
-            except ResourceGoneError:
-                # retry after refreshing the HostGAPlugin
-                GoalState.update_host_plugin_headers(self._wire_client)
-                vm_settings, vm_settings_updated = self._wire_client.get_host_plugin().fetch_vm_settings(force_update=force_update)
 
         return vm_settings, vm_settings_updated
 
-    def _fetch_extended_goal_state(self, xml_text, xml_doc, force_vm_settings_update=False):
+    def _fetch_full_wire_server_goal_state(self, incarnation, xml_doc):
         """
-        Issues HTTP requests (WireServer) for each of the URIs in the goal state (ExtensionsConfig, Certificate, Remote Access users, etc)
-        and populates the corresponding properties. If the given 'vm_settings' are not None they are used for the extensions goal state,
-        otherwise extensionsConfig is used instead.
+        Issues HTTP requests (to the WireServer) for each of the URIs in the goal state (ExtensionsConfig, Certificate, Remote Access users, etc)
+        and populates the corresponding properties.
+
+        Returns the value of ExtensionsConfig.
         """
         try:
-            logger.info('Fetching goal state [incarnation {0}]', self._incarnation)
+            logger.info('Fetching full goal state from the WireServer')
 
-            self._history.save_goal_state(xml_text)
+            role_instance = find(xml_doc, "RoleInstance")
+            role_instance_id = findtext(role_instance, "InstanceId")
+            role_config = find(role_instance, "Configuration")
+            role_config_name = findtext(role_config, "ConfigName")
+            container = find(xml_doc, "Container")
+            container_id = findtext(container, "ContainerId")
```

Comment:
This block was previously in _initialize_basic_properties(), which was merged into _fetch_full_wire_server_goal_state along with with _fetch_extended_goal_state